build_mt.py

```diff
@@ -1,599 +1,827 @@
+tree information to flag halos that intersect the lightcone and make a unique determination of which
+halo catalog epoch from which to draw the halo.
+
+Usage
+-----
+$ ./build_mt.py --help
+'''
+
+import sys
 import glob
+import time
+import gc
+import os
+from pathlib import Path
+
 import asdf
 import numpy as np
-import sys
 from scipy.interpolate import interp1d
 import matplotlib.pyplot as plt
-import numpy.lib.recfunctions as rfn
-import time
-import gc
-import os
+import numba as nb
 
 from tools.merger import simple_load, get_slab_halo, extract_superslab
 
+# these are probably just for testing; should be removed for production
+DEFAULTS = {}
+DEFAULTS['sim_name'] = "AbacusSummit_highbase_c000_ph100"  # AbacusSummit_base_c000_ph006
+DEFAULTS['z_start'] = 0.3  # 0.8#0.5
+DEFAULTS['z_stop'] = 0.65  # 1.25#0.8#0.5
+
+
 # reorder in terms of their slab number
 def reorder_by_slab(fns):
-    i_sort = np.argsort(extract_superslab(fns))
-    tmp = fns.copy()
-    for i in range(len(fns)):
-        tmp[i] = fns[i_sort[i]]
-    return tmp
+    return sorted(fns, key=extract_superslab)
+
 
 # read redshifts from merger tree files
 def get_zs_from_headers(snap_names):
     zs = np.zeros(len(snap_names))
     for i in range(len(snap_names)):
         snap_name = snap_names[i]
-        f = asdf.open(snap_name,lazy_load=True, copy_arrays=True)
-        z = np.float(f.tree['header']['Redshift'])
-        f.close()
-        zs[i] = z
+        with asdf.open(snap_name) as f:
+            zs[i] = np.float(f["header"]["Redshift"])
     return zs
 
-def distance_period(x0, x1, dimensions):
-    delta = np.abs(x0 - x1)
-    delta = np.where(delta > 0.5 * dimensions, delta - dimensions, delta)
-    return np.sqrt((delta ** 2).sum(axis=-1))
 
-def distance(x0, x1):
-    return np.sqrt(np.sum((x0-x1)**2,axis=1))
+@nb.njit
+def dist(pos1, pos2, L=None):
+    '''
+    Calculate L2 norm distance, optionally including periodicity
+    '''
+    assert pos1.shape == pos2.shape
+    N, nd = pos1.shape
+    dist = np.empty(N, dtype=pos1.dtype)
+    
+    for i in range(N):
+        delta = 0.
+        for j in range(nd):
+            dx = pos1[i][j] - pos2[i][j]
+            if L is not None:
+                if dx >= L/2:
+                    dx -= L
+                elif dx < -L/2:
+                    dx += L
+            delta += dx*dx
+        dist[i] = np.sqrt(delta)
+    return dist
+
 
 # unpack indices in Sownak's format of Nslice*1e12 + superSlabNum*1e9 + halo_position_superSlab
 def unpack_inds(halo_ids):
-
-    id_factor = 1000000000000
-    slab_factor = 1000000000
-    index = (halo_ids%slab_factor).astype(int)
-    slab_number = ((halo_ids%id_factor-index)//slab_factor).astype(int)
+    id_factor = int(1e12)
+    slab_factor = int(1e9)
+    index = (halo_ids % slab_factor).astype(int)
+    slab_number = ((halo_ids % id_factor - index) // slab_factor).astype(int)
     return slab_number, index
```

Comment:
Haven't looked at this too closely; probably fine.  I just changed to 1e12 and 1e9 so it was obvious that the counts were right.  Also, these days you can do `1_000_000_000` if you like that instead!